test/parallel/test-net-listen-fd0.js

```diff
@@ -1,20 +1,22 @@
 'use strict';
-var common = require('../common');
-var assert = require('assert');
-var net = require('net');
+const common = require('../common');
+const assert = require('assert');
+const net = require('net');
 
-var gotError = false;
+// this should fail with an async EINVAL error, not throw an exception
+net.createServer(common.fail)
+  .listen({fd: 0})
+  .on('error', common.mustCall(function(e) {
+    let gotError = false;
 
-process.on('exit', function() {
-  assert(gotError instanceof Error);
-});
+    switch (e.code) {
+      case 'EINVAL':
+      case 'ENOTSOCK':
+        gotError = e;
+        break;
+    }
 
-// this should fail with an async EINVAL error, not throw an exception
-net.createServer(common.fail).listen({fd: 0}).on('error', function(e) {
-  switch (e.code) {
-    case 'EINVAL':
-    case 'ENOTSOCK':
-      gotError = e;
-      break;
-  }
-});
+    process.on('exit', function() {
+      assert(gotError instanceof Error);
+    });
```

Comment:
I think you could get rid of the `exit` listener if you assert in the `error` callback that e.code can only be `EINVAL` or `ENOTSOCK`